### ISSUE TITLE
Full system hang after resume from suspend when tailscaled is enabled

### ISSUE THREAD

**Issue by sickill** [tailscale/tailscale#7141]
### What is the issue?

I'm seeing full system freeze / crash on my Archlinux desktop PC, so far only after suspend/resume which I do several times a day.

It doesn't happen all the time but randomly, roughly on every 3rd resume. What happens is the system wakes up, I start using it, all is good, but somewhere between 10 sec and 1 minute after resume the computer completely freezes. The screen stays on, displaying what it had at the freeze time forever, mouse/keyboard don't work, and the only thing to do is forced reset via reset button on the case.

System was super stable until about a week when 2 things happened: I installed tailscaled and also did full system upgrade (kernel, amd-ucode, amongst others). 

I did a bit of troubleshooting to pinpoint the offender:

- downgraded amd-ucode - no luck,
- downgraded kernel to several previous versions - still no luck,
- time traveled whole system to the state from a week ago (btrfs snapshot of root filesystem) - bam, all good, no freezing.

That good version of my system had previous versions of kernel and amd-ucode and had *no* tailascaled enabled nor installed.

I did a full system upgrade again to the latest, this time I skipped installation of tailscaled as I had a hunch it could be it. And yes, latest kernel and all other packages, all works fine and stable for a couple of days. To be 100% sure I installed and enabled (systemd) tailscaled last night, expecting it will cause system hangs. Today I experienced another hang. I disabled tailscaled again and all good and stable after multiple suspend/resume cycles.

Since the system is fully frozen I'm unable to check the logs (of tailscaled or dmesg) when it happens, but I can pull tailscaled logs via journalctl after system reboot. There are messages like "Temporary failure in name resolution" or "dial tcp 206.189.16.32:443: connect: no route to host" followed by success messages ("got new dial plan from control", "health("overall"): ok" etc) a couple of seconds later.

I tested on kernels: 6.0.12, 6.1.1, 6.1.4, 6.1.7, 6.1.8
I'm running Gnome 43, NetworkManager 1.40.12

### Steps to reproduce

- suspend the computer
- resume it
-  wait 10-60 sec for the system to completely freeze (happens occasionally)

### OS

Linux

### OS version

Archlinux (latest)

### Tailscale version

1.34.2

### Bug report

BUG-4e75ebd81fdc24bab891629bf9f573af5e3fc8a3ed1795a7aae77361b728cdff-20230201154842Z-6075cb77ddd3eeed

**Comment by sickill** [comment]
More info about this: sometimes when I resume the computer from sleep it doesn't freeze but the networking stops working completely and the following error messages for Intel ethernet adapter driver (igc) show up in `dmesg`:

```
[ 4508.359586] igc 0000:3c:00.0 enp60s0: Detected Tx Unit Hang
                 Tx Queue             <2>
                 TDH                  <2b>
                 TDT                  <2b>
                 next_to_use          <2b>
                 next_to_clean        <2b>
               buffer_info[next_to_clean]
                 time_stamp           <100131d71>
                 next_to_watch        <0000000061f06de1>
                 jiffies              <1001343c0>
                 desc.status          <0>
[ 4510.279478] igc 0000:3c:00.0 enp60s0: Detected Tx Unit Hang
                 Tx Queue             <2>
                 TDH                  <2b>
                 TDT                  <2b>
                 next_to_use          <2b>
                 next_to_clean        <2b>
               buffer_info[next_to_clean]
                 time_stamp           <100131d71>
                 next_to_watch        <0000000061f06de1>
                 jiffies              <100134600>
                 desc.status          <0>
[ 4512.373438] igc 0000:3c:00.0 enp60s0: Detected Tx Unit Hang
                 Tx Queue             <2>
                 TDH                  <2b>
                 TDT                  <2b>
                 next_to_use          <2b>
                 next_to_clean        <2b>
               buffer_info[next_to_clean]
                 time_stamp           <100131d71>
                 next_to_watch        <0000000061f06de1>
                 jiffies              <100134880>
                 desc.status          <0>
[ 4514.249117] igc 0000:3c:00.0 enp60s0: Detected Tx Unit Hang
                 Tx Queue             <2>
                 TDH                  <2b>
                 TDT                  <2b>
                 next_to_use          <2b>
                 next_to_clean        <2b>
               buffer_info[next_to_clean]
                 time_stamp           <100131d71>
                 next_to_watch        <0000000061f06de1>
                 jiffies              <100134ac1>
                 desc.status          <0>
[ 4516.128713] igc 0000:3c:00.0 enp60s0: Detected Tx Unit Hang
                 Tx Queue             <2>
                 TDH                  <2b>
                 TDT                  <2b>
                 next_to_use          <2b>
                 next_to_clean        <2b>
               buffer_info[next_to_clean]
                 time_stamp           <100131d71>
                 next_to_watch        <0000000061f06de1>
                 jiffies              <100134d00>
                 desc.status          <0>
```

This never happens when tailscaled is disabled.

**Comment by DentonGentry** [comment]
There was a change in 1.36 relating to the interaction with the underlying device driver, described in https://tailscale.com/blog/throughput-improvements/

Had the problem newly occurred in Tailscale 1.36.0 I'd have suspected that, but 1.34.2 does not contain the new functionality.

Nonetheless, 1.36.0 may behave differently when it comes to device driver interaction and you might want to try it.

**Comment by sickill** [comment]
Update: I'm not experiencing this problem anymore.

Since I reported it several things changed: tailscale (I'm on 1.42 now), kernel (6.2.8 now), and last but not least I replaced RAM sticks with another pair. My bet is the problem was caused by bad RAM, given I've experienced other weird behaviour on the previous sticks.

Either way, tailscale works flawlessly for me for about 2 months now. If there were no similar reports from other people then I think it's fair to just close this.
